packages/terra-show-hide/src/ShowHide.jsx

```diff
@@ -0,0 +1,157 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Toggle from 'terra-toggle';
+import IconChevronDown from 'terra-icon/lib/icon/IconChevronDown';
+import classNames from 'classnames/bind';
+import 'terra-base/lib/baseStyles';
+import styles from './ShowHide.module.scss';
+import Button from './_ShowHideButton';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * Content in the body of the component that will be shown or hidden
+   */
+  children: PropTypes.node.isRequired,
+  /**
+   * Text that will be visible to the user while the component is collapsed
+   */
+  preview: PropTypes.string.isRequired,
+  /**
+   * Button text that will be displayed when the component is collapsed
+   */
+  collapsedButtonText: PropTypes.string,
+  /**
+   * Button text that will be displayed when the component is expanded
+   */
+  expandedButtonText: PropTypes.string,
+  /**
+   * Icon displayed next to the button text
+   */
+  icon: PropTypes.element,
+  /**
+   * Sets the animation for the component when it is expanded or collapsed
+   */
+  isAnimated: PropTypes.bool,
+  /**
+   * Sets the initial state of the component to open and expanded
+   */
+  isInitiallyOpen: PropTypes.bool,
+  /**
+   * Callback function triggered when the component is collapsed
+   */
+  onClose: PropTypes.func,
+  /**
+   * Callback function triggered when the component is expanded
+   */
+  onOpen: PropTypes.func,
+};
+
+const defaultProps = {
+  collapsedButtonText: 'Show More',
+  expandedButtonText: 'Hide',
+  icon: <IconChevronDown />,
+  isAnimated: false,
+  isInitiallyOpen: false,
+};
+
+class ShowHide extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isOpen: props.isInitiallyOpen,
+    };
+
+    this.handleOnClick = this.handleOnClick.bind(this);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.isInitiallyOpen === this.props.isInitiallyOpen) {
+      return;
+    }
+
+    this.setState({ isOpen: nextProps.isInitiallyOpen });
```

Comment:
Does it make sense to call this isInitiallyOpen if it can be toggled at any time to open and close?  
If it is only for initial construction setting it the first time in the constructor should be fine.

Reply:
I copied this function off the toggle button. I commented it out to see if it only affected the isInitiallyOpen example but it turns out that it doesn't affect any of the examples. All the tests run successfully while it is commented out too. I am not sure what problem this code solved but it is not needed now. Thanks for catching that

Reply:
https://github.com/gabeparra01/terra-core/commit/3771160b712fa9ccd1f85aad70ea3f60f46db24f